src/VS.Web.CG.EFCore/ReflectedTypesProvider.cs

```diff
@@ -65,6 +65,31 @@ private CompilationResult GetCompilationResult(Compilation compilation)
             return result;
         }
 
+        public Type[] GetAllTypesInProject(bool throwOnError = false)
+        {
+            if (_compilationResult == null
+                || !_compilationResult.Success)
+            {
+                // If the compilation was not successful, just return null.
+                return null;
+            }
+
+            try
+            {
+                return _compilationResult.Assembly?.GetTypes();
+            }
+            catch (Exception ex)
+            {
+                _logger.LogMessage(ex.Message, LogMessageLevel.Error);
+                if (throwOnError)
+                {
+                    throw ex;
```

Comment:
In cases like this, I'd generally just `throw` instead of `throw ex` because `throw ex` loses the stack trace from the original exception. Either way can be appropriate, depends on what's desired.